chain/network/src/peer_store.rs

```diff
@@ -28,23 +57,48 @@ impl PeerStore {
         boot_nodes: &[PeerInfo],
     ) -> Result<Self, Box<dyn std::error::Error>> {
         let mut peer_states = HashMap::default();
+        let mut addr_peers = HashMap::default();
+
+        for mut peer_info in boot_nodes.iter().cloned() {
+            if !peer_states.contains_key(&peer_info.id) {
+                if let Some(peer_addr) = peer_info.addr.clone() {
+                    if addr_peers.contains_key(&peer_addr) {
+                        // There is already a different peer_id with this address, so don't reuse this address.
```

Comment:
If this is the case, should we just discard this peer? Does adding a peer whose addr is `None` actually do anything?

Reply:
It was a decision based on legacy code. Removing it now.